ui/brave_custom_notification/message_popup_view.cc

```diff
@@ -0,0 +1,149 @@
+  notification_(notification) {
+  SetLayoutManager(std::make_unique<views::FillLayout>());
+  views::Widget::InitParams params(views::Widget::InitParams::TYPE_POPUP);
+  params.type = views::Widget::InitParams::TYPE_WINDOW_FRAMELESS;
+  params.z_order = ui::ZOrderLevel::kFloatingWindow;
+  params.bounds = { 30, 30, 300, 100 + GetBodyHeight(notification.message())};
+#if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+  // Make the widget explicitly activatable as TYPE_POPUP is not activatable by
+  // default but we need focus for the inline reply textarea.
+  params.activatable = views::Widget::InitParams::ACTIVATABLE_YES;
+  params.opacity = views::Widget::InitParams::WindowOpacity::kOpaque;
+#else
+  params.opacity = views::Widget::InitParams::WindowOpacity::kTranslucent;
+#endif
+  params.delegate = this;
+  popup_window_ = new views::Widget();
+  popup_window_->set_focus_on_creation(true);
+  observer_.Add(popup_window_);
+
+#if defined(OS_WIN)
+  // We want to ensure that this toast always goes to the native desktop,
+  // not the Ash desktop (since there is already another toast contents view
+  // there.
+  if (!params.parent)
+    params.native_widget = new views::DesktopNativeWidgetAura(popup_window_);
+#endif
+
+  popup_window_->Init(std::move(params));
+  popup_window_->ShowInactive();
+
+  MessageView* message_view_ = MessageViewFactory::Create(notification);
+  popup_window_->SetContentsView(message_view_);
+  set_notify_enter_exit_on_child(true);
+  // g_message_popup_view = this;
+}
+
+MessagePopupView::~MessagePopupView() {
+}
+
+#if !defined(OS_MACOSX)
+float MessagePopupView::GetOpacity() const {
+  if (!IsWidgetValid())
+    return 0.f;
+  return GetWidget()->GetLayer()->opacity();
+}
+#endif
+
+void MessagePopupView::SetPopupBounds(const gfx::Rect& bounds) {
+  if (!IsWidgetValid())
+    return;
+  GetWidget()->SetBounds(bounds);
+}
+
+void MessagePopupView::SetOpacity(float opacity) {
+  if (!IsWidgetValid())
+    return;
+  GetWidget()->SetOpacity(opacity);
+}
+
+void MessagePopupView::Show() {
+}
+
+void MessagePopupView::Close() {
+  if (popup_window_ && !popup_window_->IsClosed()) {
+    popup_window_->CloseNow();
+  }
+}
+
+const char* MessagePopupView::GetClassName() const {
+  return "MessagePopupView";
+}
+
+void MessagePopupView::OnFocus() {
+  // This view is just a container, so advance focus to the underlying
+  // MessageView.
+  GetFocusManager()->SetFocusedView(message_view_);
+}
+
+void MessagePopupView::OnWidgetActivationChanged(views::Widget* widget,
+                                                 bool active) {
+  is_active_ = active;
+}
+
+void MessagePopupView::OnWidgetDestroyed(views::Widget* widget) {
+  observer_.Remove(widget);
+}
+
+bool MessagePopupView::IsWidgetValid() const {
+  return GetWidget() && !GetWidget()->IsClosed();
+}
+
+int MessagePopupView::GetBodyHeight(const base::string16& message) {
+  return (10 * (message.size() / 40)) + 10;
```

Comment:
Can we please move hardcoded values to `const` inside an empty namespace in the `.cc` file